### PR TITLE
[green-dragon-migration] add llvm coverage job

### PR DESCRIPTION
**Pull request by justice-adams-apple**
## In This PR
* Update llvm coverage job

the job has been updated to:
* Run on the correct labels
* download artifacts to S3
* Set environment variables in `environment` closure when possible
* Setup python3 venv with all dependencies to build/test
* Save coverage report as an artifact on the job rather than scp to labmaster 2

This code has been tested on https://green.lab.llvm.org/

### CODE REVIEW

**Comment by justice-adams-apple** [comment]
cc @azharudd

File: zorg/jenkins/jobs/jobs/llvm-coverage [hunk 1]
```diff
@@ -34,22 +44,51 @@ pipeline {
...
-                    sh '''
-                    python llvm-zorg/zorg/jenkins/monorepo_build.py fetch
-                    '''
+                withCredentials([string(credentialsId: 's3_resource_bucket', variable: 'S3_BUCKET')]) {
+                    sh """
+                        source ./venv/bin/activate
+                        echo "ARTIFACT=${params.ARTIFACT}"
+                        python llvm-zorg/zorg/jenkins/monorepo_build.py fetch
+                        ls $WORKSPACE/host-compiler/lib/clang/
+                        VERSION=`ls $WORKSPACE/host-compiler/lib/clang/`
```
Comment (azharudd) [inline zorg/jenkins/jobs/jobs/llvm-coverage:75]: What is this `VERSION` used for?

Comment (justice-adams-apple) [reply zorg/jenkins/jobs/jobs/llvm-coverage:75]: @azharudd 🤔 Looks like nothing. Looks like it was included in a paste job perhaps? Will remove it. Good catch

Comment (justice-adams-apple) [reply zorg/jenkins/jobs/jobs/llvm-coverage]: updated 👍

**Pull request merged**